### ISSUE TITLE
Spawn uses a lot of CPU

### ISSUE THREAD

**Issue by PeterlitsZo** [oven-sh/bun#9404]
### What version of Bun is running?

1.0.30+1424a196f

### What platform is your computer?

Linux 3.10.0-514.16.1.el7.x86_64 x86_64 unknown

### What steps can reproduce the bug?

```
$ bun repl
> import { spawn, spawnSync } from "node:child_process";
undefined
> let child = spawn("sleep", ["infinity"]);
undefined
```

Run top or btop, you can find that bun is using 100% a CPU core. You can also write a script file, and run it - same result.

When you run the `spawn` statement, it will have two process:

* `bun` - which is eating my CPU.
* `sleep` - just run and not eating my CPU.

(`Bun.spawn` and `$` also consume my one core...)

```
$ ./node-v20.11.1-linux-x64/bin/node
> const { spawn, spawnSync } = await import("node:child_process");
undefined
> let child = spawn("sleep", ["infinity"]);
undefined
```

When I use Node.js - it acts just OK - just uses 0% a CPU core.

### What is the expected behavior?

Could we let bun do not eat the total core? I think it will be better.

### What do you see instead?

Bun is eating my total core when it spawn a process.

### Additional information

Nil.

**Comment by Jarred-Sumner** [comment]
Does this reproduce outside the repl? The repl is not representative of the runtime.

**Comment by PeterlitsZo** [comment]
Of course, the problem even happens in a script file - I tried and it does still exist......

**Comment by PeterlitsZo** [comment]
<img width="923" alt="Btop" src="https://github.com/oven-sh/bun/assets/25707198/89a2568e-c0be-4738-b6d6-2105a5b94420">

Here is the result of btop on my machine. Just look at the C37 core: it just consume my total core - 100%.

Here is the command:

```
bun a.ts
```

And here is the file content:

```ts
import { spawn, spawnSync } from "node:child_process";

let child = spawn("sleep", ["infinity"]);
```

Could you reproduce? I am running it on CentOS 7. And `$` function and `Bun.spawn` function give me the same result.

**Comment by Jarred-Sumner** [comment]
> Linux 3.10.0-514.16.1.el7.x86_64 x86_64 unknown

Ah, its that. We depend on either pidfd (a linux kernel 5.15+ feature) or fallback to signalfd + a waiter thread. 

<img width="1106" alt="image" src="https://github.com/oven-sh/bun/assets/709451/d8d51bab-dd02-45aa-bf2a-3bfef53c8139">

It looks like our fallback is not idling properly

This can be reproduced on more recent linux kernels with the following 

```js
BUN_GARBAGE_COLLECTOR_LEVEL=1 BUN_FEATURE_FLAG_FORCE_WAITER_THREAD=1 bun <script>
```

**Comment by Jarred-Sumner** [comment]
This fix will be part of Bun v1.0.31

**Comment by PeterlitsZo** [comment]
Amazing! Thanks for your great work.
